src/os/ui/window.js

```diff
@@ -898,7 +898,17 @@ os.ui.WindowCtrl.prototype.disposeInternal = function() {
   }
 
   if (this.scope['modal']) {
-    this.removeModalBg();
+    const moreModalsLeft = Object.keys(os.ui.window.openWindows_).some((key) => {
+      if (key == '#' + this.scope['id']) {
+        return false;
+      }
+
+      const win = os.ui.window.getById(key);
+      const scope = win.children().first().scope();
+      return scope && scope['modal'];
+    });
+
+    this.removeModalBg(moreModalsLeft);
```

Comment:
shouldn't this only remove the modal background if there are no modals left? or are there multiple backgrounds applied?

Reply:
a background is applied for each modal.